### PR TITLE
nbft: Refactor use of nvme_msg in nbft.c

### PR DESCRIPTION
**Pull request by ikegami-t**

### CODE REVIEW

**Comment by ikegami-t** [comment]
Changes for the issue #619.

File: src/nvme/nbft.h [hunk 1]
```diff
@@ -1227,7 +1229,7 @@ struct nbft_info {
  *
  * Return: 0 on success, errno otherwise.
  */
-int nvme_nbft_read(struct nbft_info **nbft, const char *filename);
+int nvme_nbft_read(nvme_root_t r, struct nbft_info **nbft, const char *filename);
```
Comment (igaw) [inline src/nvme/nbft.h:1232]: Just passing in a `nvme_root_t` root object for printing the message is not the right thing to do. During review of this code we figured that this is not the right thing to do. That's why this argument is missing. 

We have the same problem for all non fabric functions that we don't have a `nvme_root_t` object and we might also want to print errors etc in those functions. 

The only way I see a way around this is to introduce a global variable containing the file descriptor to print to. Obviously, not totally happy with it, but what else can we do?

**Comment by ikegami-t** [comment]
Thanks for your review comments. Just fixed the patch as mentioned.

**Comment by ikegami-t** [comment]
Sorry missed to set NULL the variable after the nvme root deleted so I will update the changes later.

**Comment by igaw** [comment]
export new function, update docs, reorder free sequence

**Pull request merged**